src/levanter/optim/config.py

```diff
@@ -20,17 +20,21 @@
 @dataclass
 class OptimizerConfig(draccus.ChoiceRegistry, abc.ABC):
     learning_rate: float = 6e-4
-    weight_decay: float = 0.0
+    weight_decay: float = 0.1
 
     min_lr_ratio: float = 0.1
-    warmup_ratio: Optional[float] = None  # Deprecated. fraction of training steps to use as warmup
     """The lr scheduler operates on 4 stages: [warmup] - {[stable] - [decay]} x haps - [cooldown]"""
     warmup: float = 0.01
     """fraction of training steps to use as warmup, or steps to use. 0.0 means no warmup"""
     stable: float = 0.00
-    """fraction of training steps to use as stable, or steps to use. 0.0 means no stable"""
-    cooldown: float = 0.0
     """fraction of training steps to use as cooldown, or steps to use. 0.0 means no cooldown"""
+    rewarmup: float = 0.0
+    "If using a cycle, how much of the cycle to use as re-warmup. 0.0 means no re-warmup."
+    cooldown: Optional[float] = None
+    """Deprecated, as its semantics are confusing."""
+    cycles: int | None = None
+    """ Number of cycles to use. If None or 1, use a single cycle. Overriden by haps."""
+
     lr_schedule: str = "cosine"  # constant, cosine, linear
     haps: Optional[list[int]] = None
```

Comment:
Should we refactor this to some other name? cuz in the paper switched to wsd-s?

Reply:
yeah. that's a good idea. do you have a name in mind?

Reply:
should we just combine into the `cycles` variable? Do something like "if it's a list then... if it's an int then..."

Reply:
that's reasonable